lib/multi-level.js

```diff
@@ -135,39 +135,38 @@ class MultiLevelCache {
    */
   async _refresh(key, staleItem, updateFn, opts = {}) {
     if (!(key in this._pendingRefreshes)) {
-      try {
-        const task = updateFn(key, staleItem && staleItem.value);
-        this._pendingRefreshes[key] = task;
-
-        const value = await task;
-        const cacheItem = {
-          value,
-          expiry: Date.now() + getValue(opts, 'maxAge', this._maxAge)
-        };
-
-        const shouldCache = getValue(opts, 'shouldCache', this.shouldCache);
-        if (typeof shouldCache !== 'function') {
-          throw new TypeError('shouldCache has to be a function');
-        }
-
-        // Given that we are not ignoring this value, perform an out-of-band cache update
-        if (shouldCache(cacheItem.value)) {
-          // NB: an in-band update would `await` this Promise.all block
-          Promise.all(this._caches.map(cache => cache.set(key, cacheItem))).catch(err => {
-            throw new Error(`Error caching ${key}`, err);
-          }).then(() => {
+      this._pendingRefreshes[key] = (async () => {
```

Comment:
In short, the change is to make the clearing of the pending refreshes occur within the refresh operation instead of after.

Reply:
I like the change, but might be more readable to pull out the async function and just make this code something like:

```js
this._pendingRefreshes[key] = this._forceRefresh(key, staleItem, updateFn, opts);
```

Reply:
@msluther: love the suggestion, will do so.

Reply:
@msluther: done.